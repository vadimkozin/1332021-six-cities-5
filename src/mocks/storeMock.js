import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
