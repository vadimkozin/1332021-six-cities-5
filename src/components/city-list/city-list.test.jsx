import React from 'react';
import {Provider} from 'react-redux';
import {render} from 'enzyme';
import storeMock from '../../mocks/storeMock';
import CityList from './city-list';

describe(`CityList`, () => {
  it(`should render CityList component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <CityList />
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
