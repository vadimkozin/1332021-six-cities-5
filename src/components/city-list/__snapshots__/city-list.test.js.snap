// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList should render CityList component 1`] = `
<section
  class="locations container"
>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
        href="#"
      >
        <span>
          Cologne
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
        href="#"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
        href="#"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item false"
        href="#"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</section>
`;
